crates/solidity/inputs/language/src/definition.rs

```diff
@@ -1849,7 +1849,12 @@ codegen_language_macros::compile!(Language(
                         ),
                         Token(
                             name = Colon,
-                            definitions = [TokenDefinition(scanner = Atom(":"))]
+                            definitions = [TokenDefinition(
+                                scanner = TrailingContext(
+                                    scanner = Atom(":"),
+                                    not_followed_by = Atom("=")
```

Comment:
This is already handled by the trie codegen for the literals; I've tested this PR using previous 
```
definitions = [TokenDefinition(scanner = Atom(":"))]
```

and the codegen remains sensible and the new tests are still passing.

Could we keep the old definition to be consistent with the other tokens, please?

Reply:
Sounds great!